### PR TITLE
ipfixlolib DTLS code in own file

### PR DESCRIPTION
**Pull request by nickbroon**
Start to break down the massive ipfixlolib.c into smaller modular parts so that it's easier to understand.

No functional changes with this commit, just moving code into new files grouping data into specific structures.

Still some conditional compilation for SUPPORT_DLTS in ipfixlolib.c but it's start.

ipfixlolib_private.h is for declarations that are shared between files within ipfixlolib but not part of the public interface to the rest of Vermont.
I suspect there is more stuff in ipfixlolib.h that could be moved private header and separate source files, SCTP and RAWDIR being the obvious next candidates.

### CODE REVIEW

**Comment by ogasser** [comment]
Thanks for the PR @bluca ! I hope to find some time to review it in the next couple of weeks.

**Comment by nickbroon** [comment]
It would be great to see this merged. Is there anything I can do to help with the review?

**Comment by ogasser** [comment]
@evintila should be reviewing this PR within a week.

**Comment by evintila** [comment]
I am currently working on it.

**Comment by ogasser** [comment]
Two additional meta points that would be great to have included in this PR:
* cmake did not find openssl-1.0 on my system, I had to manually set `CMAKE_INCLUDE_PATH` and `CMAKE_LIBRARY_PATH`. Can we rewrite the *BUILDING WITH DTLS-OVER-UDP SUPPORT* section in the readme to simply suggest setting these CMAKE variables if openssl 1.0 is not found, and remove everything that says to download openssl separately since this is outdated.
* Can we add a separate build with `SUPPORT_DTLS=ON` in travis using [this approach](https://gist.github.com/jamesarosen/e29076bd81a099f0f72e#gistcomment-1478207)? This should help with future DTLS PRs.

File: src/modules/ipfix/IpfixExporterCfg.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@
  *
  */
 
+#include "common/ipfixlolib/ipfixlolib_dtls.h"
```
[the remark this answers is not in the document]
Comment (nickbroon) [reply src/modules/ipfix/IpfixExporterCfg.cpp]: Fixed.
Now builds with both `-DSUPPORT_DTLS=ON` and `-DSUPPORT_DTLS=OFF`

**Comment by nickbroon** [comment]
@ogasser  I'll take a look at fixing the readme around SSL, and perhaps even improving the cmake checks for it.

File: src/modules/ipfix/IpfixSender.cpp [hunk 2]
```diff
@@ -770,13 +777,11 @@ void IpfixSender::onSendRecordsTimeout(void) {
 		sendRecords(Always);
 	}
 }
-void IpfixSender::onBeatTimeout(void) {
```
[the remark this answers is not in the document]
Comment (nickbroon) [reply src/modules/ipfix/IpfixSender.cpp]: Fixed.

File: src/tests/ipfixlolib/example_code_2.c [hunk 3]
```diff
@@ -15,6 +15,8 @@
 #include <stdio.h>
 #include <getopt.h>
 #include "common/ipfixlolib/ipfixlolib.h"
+#include "common/ipfixlolib/ipfixlolib_dtls.h"
```
[the remark this answers is not in the document]
Comment (nickbroon) [reply src/tests/ipfixlolib/example_code_2.c]: Fixed.

File: src/tests/ipfixlolib/mtutest.c [hunk 4]
```diff
@@ -20,6 +20,8 @@
 
 #include <stdio.h>
 #include "common/ipfixlolib/ipfixlolib.h"
+#include "common/ipfixlolib/ipfixlolib_config.h"
+#include "common/ipfixlolib/ipfixlolib_dtls.h"
```
[the remark this answers is not in the document]
Comment (nickbroon) [reply src/tests/ipfixlolib/mtutest.c]: Fixed.

**Comment by ogasser** [comment]
Thanks for the PR and review!

**Pull request merged**